xbmc/platform/darwin/osx/SDLMain.mm

```diff
@@ -353,54 +374,142 @@ - (void) applicationDidFinishLaunching: (NSNotification *) note
     data1: 0
     data2: 0];
   //
-  [NSApp postEvent: event atStart: true];
+  [NSApp postEvent: event atStart: YES];
 }
 
-/*
- * Catch document open requests...this lets us notice files when the app
- *  was launched by double-clicking a document, or when a document was
- *  dragged/dropped on the app's icon. You need to have a
- *  CFBundleDocumentsType section in your Info.plist to get this message,
- *  apparently.
- *
- * Files are added to gArgv, so to the app, they'll look like command line
- *  arguments. Previously, apps launched from the finder had nothing but
- *  an argv[0].
- *
- * This message may be received multiple times to open several docs on launch.
- *
- * This message is ignored once the app's mainline has been called.
- */
-- (BOOL)application:(NSApplication *)theApplication openFile:(NSString *)filename
+- (NSArray*) fileListFromPasteboard:(NSPasteboard*)pboard
 {
-  const char *temparg;
-  size_t arglen;
-  char *arg;
-  char **newargv;
+  NSArray* pboardItems = pboard.pasteboardItems;  // copy as soon as you can, may change, disappear etc...
+  NSMutableArray* foundFiles = [[[NSMutableArray alloc] init] autorelease];
+  
+  for (NSPasteboardItem* item in pboardItems)
+  {
+    for (NSString* type in item.types)
+    {
+      if ([type isEqualToString:@"public.file-url"])
+      {
+        NSString* string = [item stringForType:type];
+        
+        NSDataDetector* linkDetector = [NSDataDetector dataDetectorWithTypes:NSTextCheckingTypeLink error:nil];
+        NSArray* matches = [linkDetector matchesInString:string options:0 range:NSMakeRange(0, [string length])];
+        
+        if (matches.count)
+        {
+          // Enumerat all of the links and open one by one
+          for (NSTextCheckingResult* nextMatch in matches)
+          {
+            NSString* substringForMatch = [string substringWithRange:nextMatch.range];
+            
+            if (substringForMatch)
+            {
+              NSURL* foundURL = [NSURL URLWithString:substringForMatch];
+              NSString* fileName = nil;
+              
+              if (foundURL.isFileReferenceURL || [foundURL.scheme caseInsensitiveCompare:@"file"] == NSOrderedSame)   // Already a file system path URL
+                fileName = foundURL.path;
+              else
+              {
+                foundURL = [NSURL fileURLWithPath:substringForMatch];
```

Comment:
It's been a decade i've added this and at the first sight even i could not spot the difference
this is the difference, first trying to look for a match using URLWithString: second it uses fileURLWithPath: